src/main/kotlin/carracing/CarRace.kt

```diff
@@ -1,24 +1,29 @@
 package carracing
 
-import kotlin.random.Random
+import carracing.RandomNumberGenerator.getRandom
+import carracing.view.ResultView
 
-class CarRace(val numCars: Int, val numRounds: Int) {
-    val cars = generateCars()
+class CarRace(val cars: List<Car>, val numRounds: Int) {
 
-    fun startRace() {
+    init {
         require(cars.size >= MIN_NUMBER_OF_CARS) { "Not enough cars to race" }
         require(numRounds > MIN_NUMBER_OF_ROUNDS) { "Select at least one round to start the race" }
+    }
+
+    fun startRace() {
         repeat(numRounds) {
             cars.forEach { it.move(random = getRandom()) }
             ResultView.showStatus(cars)
         }
+        ResultView.displayWinners(winners = getWinners())
     }
 
-    private fun generateCars() = List(numCars) { Car(name = "Car ${it + 1}") }
-    private fun getRandom() = Random.nextInt(RANDOM_NUMBER_BOUNDARY)
+     fun getWinners(): List<Car> {
+        val maxPosition = cars.maxOf { it.position }
+        return cars.filter { it.position == maxPosition }
```

Comment:
It looks like this code is pulling out each car’s position to determine the winners externally.
Consider refactoring so that the logic expresses _tell the cars to check if they are winners_, rather than _ask the cars for their position and figure it out externally_.

Here’s a reference you might find helpful:
- [Tell Don't Ask](https://martinfowler.com/bliki/TellDontAsk.html)